app/models/pubmed_source_record.rb

```diff
@@ -99,6 +99,61 @@ def extract_mesh_headings_from_pubmed_record(pubmed_record)
     mesh_headings_for_record
   end
 
+  # see https://dtd.nlm.nih.gov/ncbi/pubmed/doc/out/180101/el-Year.html
+  def extract_year_from_pubmed_record(publication)
+    year = nil
+
+    # look for a year in all of the xpath locations in order
+    #  stop after the first produces something that looks like a year
+    pubmed_date_xpaths('Year').each do |path|
+      match = publication.xpath(path).text.match(/[12][0-9]{3}/)
+      next unless match
+
+      year = match.to_s
+      break
+    end
+
+    year
+  end
```

Comment:
@peetucket I wonder if it's worth simplifying this method (and its buddies below) to something more like:

```suggestion
potential_years.find { |year| year.match?(/[12][0-9]{3}/) }.to_s
  # see https://dtd.nlm.nih.gov/ncbi/pubmed/doc/out/180101/el-Year.html
  def extract_year_from_pubmed_record(publication)
    pubmed_date_xpaths('Year').find { |path| publication.xpath(path).text.match?(/[12][0-9]{3}/) }.to_s.presence
  end
```

Reply:
Thanks for the suggestion.  I tweaked slightly to make it work (this wasn't quite working as expected), and applied a similar change to the other day and month methods.